bibo/models.py

```diff
@@ -0,0 +1,3 @@
+import collections
+
+SearchResult = collections.namedtuple("SearchResult", ["entry", "match"])
```

Comment:
Maybe you'll disagree, but `_nested_append` and `_update_result` from `bibo.query` could move here (or somewhere similar, but not `bibo query`) to leave search/matching for that module. If we need a bit more wrapped up in `SearchResult`, maybe its time for `attrs`: https://www.attrs.org/en/stable/why.html#namedtuples

Reply:
I generally prefer not to mix logic and data structures too much. One reason is imports. If `models.py` doesn't depend on any `internals` functionality it is easier to import. Will be happy to restructure some of `query` / `internals` in a more sensible way though. Regarding `attrs`, would check this out. Don't see the benefit at the moment, as our use case is very simple.